### PR TITLE
feat: new blockchain docs

### PR DESCRIPTION
**Pull request by alenmestrov**
Created new version of blockchain integrations documentation that reflects latest changes of integrations.

created based on this document:

https://docs.google.com/document/d/1pxsKzKBRUDzTBQ0IE0kVK2_0BcBS29dO-JSzEeEbNEE/edit?tab=t.0

Screenshot 2024-12-06 at 11 55 24
Screenshot 2024-12-06 at 11 55 35
Screenshot 2024-12-06 at 11 55 43
Screenshot 2024-12-06 at 13 02 33

### CODE REVIEW

**Comment by chefsale** [comment]
Text, text, text. Can we please draw some graphs?

File: docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+Calimero currently supports the following blockchain protocols:
+
+- NEAR
+- Starknet
+
+## Supported Operations
+
+- Read
+  - Read the data from the Calimero contracts
```
Comment (chefsale) [inline docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx:22]: Read and write data to any smart contract?

File: docs/03-core-concepts/06-blockchains-integration/02-context-contract.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+## Setup
+
+To create context on specific blockchain user will need to add the `--protocol` flag to the `calimero-cli` when initializing the node as explained in the [Setup](/docs/02-getting-started/01-setup) section.
+
+<Tabs
+    defaultValue="ssh"
+    values={[
+        {label: 'Starknet', value: 'starknet'},
```
Comment (chefsale) [inline docs/03-core-concepts/06-blockchains-integration/02-context-contract.mdx:21]: NEAR, ICP?

Comment (alenmestrov) [reply docs/03-core-concepts/06-blockchains-integration/02-context-contract.mdx:21]: For NEAR we have a note underneath that says that the flag is not needed.

File: docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx (hunk 1 above)
Comment (alenmestrov) [reply docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx:22]: Currently, we are supporting the write to any contract through the cross-contract call in Proxy contract. But for reading it was agreed that in the case of reading the data from the blockchain (except the data from Context and Proxy contracts) developers will need to implement it themselves

Comment (chefsale) [reply docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx:22]: Why is that not written here then?

Comment (chefsale) [reply docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx:22]: Most important information on the top, aka the part of how do you interact with the blockchain. Reading and writing data to calimero contract doesn't say anything, leave limitations here, but moved the relevant parts to the top and draw a diagram of how the flow looks like so people can get the basic understanding under a minute, then expand with specifics later on.

Comment (chefsale) [reply docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx:22]: Aka the question you want to answer is what does reading and writing data mean. Calling contract, etc.

**Comment by alenmestrov** [comment]
> Text, text, text. Can we please draw some graphs?

@chefsale added screenshots of graphs, let me know what you think

**Comment by alenmestrov** [comment]
Added draft doc for Proxy contract intregration/interaction tutorial

File: docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+- Configuration of the internal configuration variables
+- Proposal creation and execution
+
+```mermaid
+graph TD
+    A[Proxy Contract] -->|Cross-contract Calls| B[Other Contracts]
+    A -->|Transfers| C[Token Transfers]
+    A -->|Data Storage| D[Key-Value Storage]
+    A -->|Configuration| E[Internal Variables]
+    E -->|Change Votes Required| F[Vote Configuration]
```
Comment (MatejVukosav) [inline docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx:69]: `Change Votes Required` it is not clear what is this label for

Comment (alenmestrov) [reply docs/03-core-concepts/06-blockchains-integration/01-blockchain-integration-general.mdx:69]: Extended description in Functionality section for configuration variables

**Review by MatejVukosav — APPROVED** [review]

**Comment by MatejVukosav** [comment]
@alenmestrov can you resolve this?

**Pull request merged**